localstack/services/apigateway/integration.py

```diff
@@ -0,0 +1,49 @@
+import json
+import logging
+
+from localstack import config
+from localstack.services.apigateway.apigateway_listener import apply_template
+from localstack.services.apigateway.context import ApiInvocationContext
+from localstack.utils.aws import aws_stack
+from localstack.utils.common import make_http_request, to_str
+
+LOG = logging.getLogger(__name__)
+
+
+class BackendIntegration:
+    """
+    Backend integration
+    """
+
+
+class SnsIntegration(BackendIntegration):
+    __slots__ = ["invocation_context"]
+
+    def __init__(self, invocation_context: ApiInvocationContext):
+        self.invocation_context = invocation_context
+
+    def invoke(self):
+        try:
+            data = self.invocation_context.data
+            data = json.dumps(data) if isinstance(data, (dict, list)) else to_str(data)
+            payload = apply_template(
+                self.invocation_context.integration,
+                "request",
+                data,
+                path_params=self.invocation_context.path_params,
+                query_params=self.invocation_context.query_params(),
+                headers=self.invocation_context.headers,
+            )
+        except Exception as e:
+            LOG.warning("Failed to apply template for SNS integration", e)
+            raise
+        uri = (
+            self.invocation_context.integration.get("uri")
```

Comment:
Same as above - could this become a very common pattern that we could reuse in multiple places? (i.e., adding a util to `ApiInvocationContext`)

Reply:
ditto, I'll clean it up over time!